alvr/client_openxr/src/extra_extensions/face_tracking2_fb.rs

```diff
@@ -0,0 +1,114 @@
+use alvr_common::{anyhow::Result, ToAny};
+use openxr::{self as xr, raw, sys};
+use std::ptr;
+
+impl super::ExtraExtensions {
+    pub fn supports_fb_visual_face_tracking(
+        &self,
+        instance: &xr::Instance,
+        system: xr::SystemId,
+    ) -> bool {
+        self.get_props(instance, system, unsafe {
+            sys::SystemFaceTrackingProperties2FB::out(ptr::null_mut()).assume_init()
+        })
+        .map(|props| props.supports_visual_face_tracking.into())
+        .unwrap_or(false)
+    }
+
+    pub fn supports_fb_audio_face_tracking(
+        &self,
+        instance: &xr::Instance,
+        system: xr::SystemId,
+    ) -> bool {
+        self.get_props(instance, system, unsafe {
+            sys::SystemFaceTrackingProperties2FB::out(ptr::null_mut()).assume_init()
+        })
+        .map(|props| props.supports_audio_face_tracking.into())
+        .unwrap_or(false)
+    }
+
+    pub fn create_face_tracker2_fb<G>(
+        &self,
+        session: &xr::Session<G>,
+        visual: bool,
+        audio: bool,
+    ) -> Result<FaceTracker2FB> {
+        let ext_fns = self.ext_functions_ptrs.fb_face_tracking2.to_any()?;
+
+        let mut requested_data_sources = vec![];
+        if visual {
```

Comment:
is it really a safe wrapper if it doesn't check if support is actually present here too? :eye:

Reply:
(what I mean is that it's technically safe for the purposes of memory safety, but not safe for the purposes of openxr compliance)

Reply:
WSYM? Support for visual vs audio? if support is not present this function will return an error, without crashing.

Reply:
I forgot that that's the way openxr works